bin/liquidator/src/rebalance.rs

```diff
@@ -222,47 +249,93 @@ impl Rebalancer {
         Ok((sig, route))
     }
 
-    async fn determine_best_jupiter_tx(
+    fn get_alternative_token_amount(
         &self,
-        full: &jupiter::Quote,
-        alternatives: &[jupiter::Quote],
-    ) -> anyhow::Result<(TransactionBuilder, jupiter::Quote)> {
-        let builder = self
-            .mango_client
-            .jupiter()
-            .prepare_swap_transaction(full)
-            .await?;
-        let tx_size = builder.transaction_size()?;
-        if tx_size.is_within_limit() {
-            return Ok((builder, full.clone()));
-        }
-        trace!(
-            route_label = full.first_route_label(),
-            %full.input_mint,
-            %full.output_mint,
-            ?tx_size,
-            limit = ?TransactionSize::limit(),
-            "full route does not fit in a tx",
-        );
+        in_token_index: &u16,
+        in_amount_quote: u64,
+    ) -> anyhow::Result<(Pubkey, u64)> {
+        let in_token: &TokenContext = self.mango_client.context.token(*in_token_index);
+        let in_price = self
+            .account_fetcher
+            .fetch_bank_price(&in_token.first_bank())?;
+        let in_amount = (I80F48::from(in_amount_quote) / in_price)
+            .ceil()
+            .to_num::<u64>();
+
+        Ok((in_token.mint, in_amount))
+    }
 
-        if alternatives.is_empty() {
-            anyhow::bail!(
-                "no alternative routes from {} to {}",
-                full.input_mint,
-                full.output_mint
+    async fn determine_best_swap_tx(
+        &self,
+        mut routes: Vec<swap::Quote>,
+        input_mint: Pubkey,
+        output_mint: Pubkey,
+    ) -> anyhow::Result<(TransactionBuilder, swap::Quote)> {
+        let mut prices = HashMap::<Pubkey, I80F48>::new();
+        let mut get_or_fetch_price = |m| {
+            prices.get(&m).copied().unwrap_or_else(|| {
```

Comment:
the more idiomatic way to write this would be with `prices.entry(..)`.

this optimization could also be somewhat premature, since fetch_bank_price shouldn't be very expensive.